examples/phyast.py

```diff
@@ -0,0 +1,390 @@
+              s += "block("
+              for j in range(len(a.orelse)):
+                  aa = a.orelse[j]
+                  if j > 0:
+                      s += ","
+                  if j + 1 == len(a.orelse):
+                      s += self.recompile(aa,allowreturn)
+                  else:
+                      s += self.recompile(aa)
+              s += ")"
+          else:
+              s += self.recompile(a.orelse[0],allowreturn)
+          s += ")"
+          return s
+      elif nm == "Compare":
+          args = [arg for arg in ast.iter_child_nodes(a)]
+          sym = "?"
+          nn = args[1].__class__.__name__
+          if nn == "Lt":
+              sym = "<"
+          elif nn == "Gt":
+              sym = ">"
+          elif nn == "LtE":
+              sym = "<="
+          elif nn == "GtE":
+              sym = ">="
+          elif nn == "NotEq":
+              sym = "!="
+          elif nn == "Eq":
+              sym = "=="
+          return self.recompile(args[0])+sym+self.recompile(args[2])
+      elif nm == "UnaryOp":
+          args = [arg for arg in ast.iter_child_nodes(a)]
+          nm2 = args[0].__class__.__name__
+          nm3 = args[1].__class__.__name__
+          if nm2 == "USub":
+            if nm3 == "BinOp" or self.groupAggressively:
+                return "-(" + self.recompile(args[1])+")"
+            else:
+                return "-" + self.recompile(args[1])
+          else:
+            raise Exception(nm2)
+      else:
+          raise Exception(nm)
+
+local_namespace = {}
+# Create the decorator
+class phylanx(object):
+    def __init__(self,f):
+        global local_namespace
+        self.f = f
+        # Get the source code
+        src = inspect.getsource(f) #getsource(f)
+        # Before recompiling the code, take
+        # off the decorator from the source.
+        src = re.sub(r'^\s*@\w+\s*','',src)
+
+        # Create the AST
+        tree = ast.parse(src)
+        #astdump.indented(tree)
+        #get_info(tree)
+        #globals()["fn"]=
+        r = Recompiler()
+        self.new_src = "block(" + r.recompile(tree)+')\n'
+        print('new_src =',self.new_src)
+        lisp_fmt(self.new_src)
+        # Compile the AST
+        #tmp_name = "new_func_name"
+        #self.f1=compile(tmp_name+"="+new_src,filename='<string>',mode='single')
+        # Exec creates the function in the namespace
+        #exec(self.f1,{},local_namespace)
+        # Get the function from the namespace
+        #self.nf = local_namespace[tmp_name]
+
+    def __call__(self,*args):
+        #global local_namespace
+        #print(local_namespace)
+        #print("Recompiled...")
+        #return self.nf(*args)
+        #print('len=',len(args))
+        if len(args)==1:
+            return et.phylisp_eval(self.new_src,args[0])
+        elif len(args)==2:
+            return et.phylisp_eval(self.new_src,args[0],args[1])
+        else:
+            return et.phylisp_eval(self.new_src,*args)
+
+@phylanx
```

Comment:
Could you please separate the implementation of the decorator from the tests?

Reply:
That should be done now.